src/compiler/scanner.ts

```diff
@@ -1828,6 +1831,9 @@ namespace ts {
                         pos++;
                         return token = SyntaxKind.GreaterThanToken;
                     case CharacterCodes.question:
+                        if (text.charCodeAt(pos + 1) === CharacterCodes.question) {

```

Comment:
Nit: If you re-order the `pos++` before the `if`, the code become marginally simpler.

Reply:
I prefer to be consistent with the existed code.